framework/di/Instance.php

```diff
@@ -128,7 +128,11 @@ public static function ensure($reference, $type = null, $container = null)
         }
 
         if ($reference instanceof self) {
-            $component = $reference->get($container);
+            try {
```

Comment:
try-catch may slow down overall execution and since it's a central part of the framework executed hundreds of times per request I think we need to test how it affects performance. @cebe can you do it?